### PR TITLE
Remove obsolete Migrations and RevEng APIs

### LINKED ISSUE #13292: RevEng: Remove infra around underlyingstoretype which was to support type aliases
With #12633 type alises are going away from DatabaseModel level. i.e. providers will find out underlying type for aliased type and swap them out so EF never sees aliases. The infra to support aliases in EF Core design is public and should be removed.

### PR DESCRIPTION
**Pull request by bricelam**
Resolves #13292, part of #12835

### CODE REVIEW

**Comment by smitpatel** [comment]
>         ///         This method does not call the overload of ColumnDefinition that takes fixedLength because doing so

You may want to update this

---
Refers to: src/EFCore.Relational/Migrations/MigrationsSqlGenerator.cs:1208 in 3fef18f. [](commit_id = 3fef18f6bdd35ef427bf3681df9a20bae4dd6ace, deletion_comment = False)

File: test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs [hunk 1]
```diff
@@ -313,20 +313,16 @@ public void Do_not_use_database_names_for_columns()
         }
 
         [Theory]
-        [InlineData("nvarchar(450)", null, null)]
-        [InlineData("alias for string", "nvarchar(450)", "alias for string")]
-        public void Column_type_annotation(string StoreType, string underlyingType, string expectedColumnType)
+        [InlineData("nvarchar(450)", null)]
+        [InlineData("alias for string", "alias for string")]
```
Comment (smitpatel) [inline test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs]: Expected column type will never be aliased type.

File: test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs [hunk 2]
```diff
@@ -1771,18 +1767,14 @@ public void Correct_arguments_to_scaffolding_typemapper()
         }
 
         [Fact]
-        public void Unmapped_column_with_underlying_store_type_is_ignored()
+        public void Unmapped_column_is_ignored()
```
Comment (smitpatel) [inline test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs]: This test may be redundant. not sure

File: test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs (hunk 1 above)
Comment (bricelam) [reply test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs]: Updated to verify that not-by-convention types are set.

File: test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs (hunk 2 above)
Comment (bricelam) [reply test/EFCore.Design.Tests/Scaffolding/Internal/RelationalScaffoldingModelFactoryTest.cs]: Verified not redundant

**Pull request merged**